src/main/java/org/sagebionetworks/web/client/widget/entity/tabs/AbstractTablesTab.java

```diff
@@ -284,15 +283,10 @@ public void setTargetBundle(EntityBundle bundle, Long versionNumber) {
 		modifiedCreatedBy.setVisible(false);
 		view.setTableUIVisible(isShownInTab);
 		view.setActionMenu(tab.getEntityActionMenu());
-		tab.getEntityActionMenu().setTableDownloadOptionsVisible(isShownInTab);
-		boolean isCurrentVersion = version == null;
-
-		if (isDataset && isCurrentVersion) {
-			// SWC-5878 - On the current (non-snapshot) version of a dataset, only editors should be able to download
-			tab.getEntityActionMenu().setTableDownloadOptionsEnabled(bundle.getPermissions().getCanCertifiedUserEdit());
-		} else {
-			tab.getEntityActionMenu().setTableDownloadOptionsEnabled(true);
+		if (!isShownInTab) {
+			tab.getEntityActionMenu().setTableDownloadOptionsVisible(false);
 		}
```

Comment:
Hide the download options if we're no longer looking at a specific Table. Previously, we were doing this:

```java
tab.getEntityActionMenu().setTableDownloadOptionsVisible(isShownInTab);
```

but we no longer want to unconditionally show the download options if we're looking at a Table. If we don't query the table (e.g. because there are no columns), then we shouldn't allow Downloads.